src/simulation/m_hyperelastic.fpp

```diff
@@ -0,0 +1,336 @@
+!>
+!! @file m_hyperelastic.f90
+!! @brief Contains module m_hyperelastic
+
+#:include 'macros.fpp'
+
+!> @brief This module consists of subroutines used in the calculation
+!!              of the cauchy tensor
+
+module m_hyperelastic
+
+    ! Dependencies =============================================================
+
+    use m_derived_types        !< Definitions of the derived types
+
+    use m_global_parameters    !< Definitions of the global parameters
+
+    use m_variables_conversion !< State variables type conversion procedures
+
+    use m_finite_differences
+
+    ! ==========================================================================
+
+    implicit none
+
+    private; public :: s_hyperelastic_rmt_stress_update, &
+ s_initialize_hyperelastic_module, &
+ s_finalize_hyperelastic_module
+
+    !> @name Abstract interface for creating function pointers
+    !> @{
+    abstract interface
+
+        !> @name Abstract subroutine for the infinite relaxation solver
+        !> @{
+        subroutine s_abstract_hyperelastic_solver(btensor, q_prim_vf, G, j, k, l)
```

Comment:
we have gotten rid of these `abstract` function pointers because they cause problems on Frontier/Cray/AMD GPUs. instead we just use a wrapper subroutine that uses if statements to select the right solver (see the new m_riemann_solvers module and how it selects between HLL and HLLC)

Reply:
This abstract interface doesn't appear to be used, so I deleted it.